src/main/java/org/apache/mesos/offer/MesosResourcePool.java

```diff
@@ -113,8 +113,16 @@ private void releaseAtomicResource(MesosResource mesRes) {
 
   private MesosResource consumeReserved(ResourceRequirement resReq) {
     MesosResource mesRes = reservedPool.get(resReq.getResourceId());
+
     if (mesRes != null) {
-      reservedPool.remove(resReq.getResourceId());
+      if (mesRes.isAtomic() &&
```

Comment:
Couldn't hurt to add a quick comment to the 'return null' case explaining the 'why'